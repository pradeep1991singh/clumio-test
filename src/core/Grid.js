import React from "react";
import Card from "./Card";

const Grid = ({ items }) => {
  return (
    <div className="dark:bg-gray-900 p-10 grid gap-10 grid-cols-4">
      {items.map((item) => (
        <div key={item.id}>
          <Card {...item} />
        </div>
      ))}
      {!items.length && (
        <span className="font-semibold text-xl tracking-tight dark:text-gray-300">
          No Result found!
        </span>
      )}
    </div>
  );
};

export default Grid;
