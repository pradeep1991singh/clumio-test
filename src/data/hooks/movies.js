import { useQuery } from "react-query";
import { filterList } from "../../services/utils";
import {
  getPopular,
  getTrending,
  getNewest,
  getTopRated,
} from "../api/movieApi";
import { useGetGenreList } from "./genres";

export const useGetPopularList = (filterType, filter) => {
  const { data: genres } = useGetGenreList();
  return useQuery("popularList", getPopular, {
    select: (data) => filterList(data, filterType, filter, genres), // debounce this
    refetch: true,
  });
};

export const useGetTrendList = (filterType, filter) => {
  const { data: genres } = useGetGenreList();
  return useQuery("trendList", getTrending, {
    select: (data) => filterList(data, filterType, filter, genres), // debounce this
    refetch: true,
  });
};

export const useGetNewestList = (filterType, filter) => {
  const { data: genres } = useGetGenreList();
  return useQuery("newestList", getNewest, {
    select: (data) =>
      filterList({ results: [data] }, filterType, filter, genres), // debounce this
    refetch: true,
  });
};

export const useGetTopRatedList = (filterType, filter) => {
  const { data: genres } = useGetGenreList();
  return useQuery("topRatedList", getTopRated, {
    select: (data) => filterList(data, filterType, filter, genres), // debounce this
    refetch: true,
  });
};
