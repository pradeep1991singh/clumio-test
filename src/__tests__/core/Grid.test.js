import React from "react";
import { render } from "@testing-library/react";
import Grid from "../../core/Grid";
import { moviesMock } from "../../__mocks__/movies.mock";

describe("<Grid />", () => {
  test("Snapshot", () => {
    const { asFragment } = render(<Grid items={moviesMock} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
