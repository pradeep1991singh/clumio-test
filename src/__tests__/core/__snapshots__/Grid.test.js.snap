// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> Snapshot 1`] = `
<DocumentFragment>
  <div
    class="dark:bg-gray-900 p-10 grid gap-10 grid-cols-4"
  >
    <div>
      <div
        class="max-w-sm overflow-hidden"
      >
        <img
          alt="Card"
          class="w-full"
          src="example.com"
        />
        <div
          class="px-6 py-4"
        >
          <div
            class="font-bold text-xl mb-2 text-blue-500"
          >
            title
          </div>
          <p
            class="text-gray-700 text-base dark:text-gray-500"
          >
            genre1genre2, 2022
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
